src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -8114,6 +8114,126 @@ public static ulong Splice(ulong[] first, ulong[] second, ulong[] maskArray, int
+        }
+
+        private static T GetGatherVectorResultByIndex<T, ExtendedElementT, Index>(int index, T[] firstOp, ExtendedElementT[] secondOp, Index[] thirdOp)
+                where T : INumberBase<T> 
+                where ExtendedElementT : INumberBase<ExtendedElementT> 
+                where Index : IBinaryInteger<Index>
+        {
+            return (firstOp[index] == T.Zero) ? T.Zero : T.CreateTruncating(secondOp[int.CreateChecked(thirdOp[index])]);
+        }
+
+        private static bool CheckGatherVectorBehaviorCore<T, ExtendedElementT, Index>(T[] firstOp, ExtendedElementT[] secondOp, Index[] thirdOp, T[] result, Func<int, T, T> map) 
+                where T : INumberBase<T> 
+                where ExtendedElementT : INumberBase<ExtendedElementT> 
+                where Index : IBinaryInteger<Index>
+        {
+            for (var i = 0; i < firstOp.Length; i++)
+            {
+                T gatherResult = GetGatherVectorResultByIndex(i, firstOp, secondOp, thirdOp);
+                gatherResult = map(i, gatherResult);
+                if (result[i] != gatherResult)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public static bool CheckGatherVectorBehavior<T, ExtendedElementT, Index>(T[] firstOp, ExtendedElementT[] secondOp, Index[] thirdOp, T[] result) 
+                where T : INumberBase<T> 
+                where ExtendedElementT : INumberBase<ExtendedElementT> 
+                where Index : IBinaryInteger<Index>
+        {
+            return CheckGatherVectorBehaviorCore(firstOp, secondOp, thirdOp, result, (_, gatherResult) => gatherResult);
+        }
+
+        public static bool CheckGatherVectorBehavior<T, ExtendedElementT, Index>(T[] maskOp, T[] firstOp, ExtendedElementT[] secondOp, Index[] thirdOp, T[] falseOp, T[] result) 
+                where T : INumberBase<T> 
+                where ExtendedElementT : INumberBase<ExtendedElementT> 
+                where Index : IBinaryInteger<Index>
+        {
+            return CheckGatherVectorBehaviorCore(firstOp, secondOp, thirdOp, result, (i, gatherResult) => ConditionalSelectResult(maskOp[i], gatherResult, falseOp[i]));
+        }
+
+        private static bool CheckFirstFaultingBehaviorCore<T>(T[] result, Vector<T> faultResult, Func<int, bool> checkIter) where T : INumberBase<T>
+        {
+            bool hitFault = false;
+
+            for (var i = 0; i < result.Length; i++)
+            {
+                if (hitFault)
+                {
+                    if (faultResult[i] != T.Zero)
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    if (faultResult[i] == T.Zero)
+                    {
+                        // There has to be a valid value for the first element, so check it.
```

Comment:
Why? Is that because you forcefully make sure that at least first element is valid? Should we add coverage to pass address such that all elements including the first element is getting loaded from an invalid address?

Reply:
I see. Can you add a test to cover the scenario then? It should be similar to `RunBasicScenario_NonFaulting` except that you pass `TrueAll` or something and make sure we get exception?